assets/js/blocks/cart-checkout/checkout/edit.js

```diff
@@ -59,7 +60,8 @@ const BlockSettings = ( { attributes, setAttributes } ) => {
 	// setting initial password.
 	// Also implicitly gated to feature plugin, because Checkout
 	// block is gated to plugin
-	const showCreateAccountOption = compareWithWooVersion( '4.7.0', '<=' );
+	const showCreateAccountOption =
+		CHECKOUT_ALLOWS_SIGNUP && compareWithWooVersion( '4.7.0', '<=' );
```

Comment:
This option is now potentially confusing if the store has checkout disabled. For example, if store has checkout signup disabled, this option won't work. I don't think that's a blocker for this PR, but we might want to tweak the UI or add a warning message in future.

Reply:
I agree, but in the absence of that UI treatment, I think it's preferable to _not_ show the option since it is ineffective if signup is globally disabled. So I think this is a safer starting point. I think it'd be actually more confusing if the option shows when signup is disabled globally.

Reply:
Works for me :) In testing it seems pretty reasonable.